docs/src/features/error-types.md

```diff
@@ -7,73 +7,151 @@ lang: en-US
+
+```
+=UNKOWN_FUNCTION(A1)
+```
+
+This indicates an error in your spreadsheet logic.
 
 ### **`#REF!`**
 
-**Cause:** Invalid cell reference, often from deleting cells used in a formula.  
-**Fix:** Update the formula with correct references.
+Indicates an invalid cell reference, often from deleting cells used in a formula.
+
+They can appear as a result of a computation or in a formula. Examples:
+
+```
+=Sheet34!A1
+```
+
+If `Sheet34` doesn't exist it will return `#REF!`
+
+This is a genuine error. It indicates that part of your formula references a cell or range that is missing.
 
 ### **`#NUM!`**
 
-**Cause:** Invalid numeric operation (e.g., calculating a square root of a negative number).  
-**Fix:** Adjust the formula to ensure valid numeric operations.
+Invalid numeric operation (e.g., calculating a square root of a negative number).  
+Adjust the formula to ensure valid numeric operations.
+
+Sometimes a `#NUM!` might be expected signalling the user that some parameter is out of scope.
 
 ### **`#N/A`**
 
-**Cause:** A value is not available, often in lookup functions like VLOOKUP.  
-**Fix:** Ensure the lookup value exists or use IFNA() to handle missing values:
+A value is not available, often in lookup functions like VLOOKUP.
+
+This is frequnly not an error in your spreadsheetlogic.
+
+You can produce a prettier answer using the [`IFNA`](/functions/information/isna) formula:
 
 ```
 =IFNA(VLOOKUP(A1, B1:C10, 2, FALSE), "Not Found")
 ```
 
 ### **`#NULL!`**
 
-**Cause:** Incorrect range operator in a formula (e.g., missing a colon between cell references).  
-**Fix:** Use correct range operators (e.g., A1:A10).
+Incorrect range operator in a formula (e.g., missing a colon between cell references).
 
+### **`#SPILL!`**
+
+A cell in a formula will overwrite content in other cells.
+This cannot happen riht now in IronCalc as formulas don't spill yet.
 
 ### **`#CIRC!`**
 
-**Cause:** Circular reference.  
-**Fix:** Remove the circular reference.
+Circular reference. This is an error is your spreadsheet and must be fixed.
+Means that during teh course of a computation a circular dependency was found.
+
+A circular dependency is a dependency of a formula on itself.
+
+For instance in the cell `A1` the formula `=A1*2` is a circular dependency.
+
+Other spreadsheet engines use circular dependencies to do "loop computations", run "sensitivity analysis" or "goal seek".
+
+IronCalc doesn't support any of those at the moment.
+
+## IronCalc specific errors
+
+---
+
+### **`#ERROR!`**
+
+General formula issue, like syntax errors or invalid references.
+In general Excel does not let you enter incorrect formulas but IronCalc will.
+
+This will make your workbook imcompatible with Excel
+
+For instace an incomplete formula
+
+```
+=A1+
+```
+
+### **`#N/IMPL!`**
+
+A particular feature is not yet implemented in IronCalc
+
+Look if there is a Github ticket or contact us via email, Discord or bluesky
```

Comment:
I think I would leave out bluesky, email or Discord is fine. But I would add a link to both.